src/python/grapl_analyzerlib/grapl_analyzerlib/grapl_client.py

```diff
@@ -17,6 +18,16 @@ class GraphClient(DgraphClient):
     def from_host_port(cls, host: str, port: int) -> "GraphClient":
         return cls(*(DgraphClientStub(f"{host}:{port}"),))
 
+    @contextmanager
+    def txn_context(self, read_only: bool = False) -> Iterator[Txn]:
```

Comment:
I think a minor comment here explaining the behavior might make sense. Also, this like... really really works right? `finally` is so nutty and I've never considered how it works with a yield so just checking lol